build-farm/platform-specific-configurations/downloaders.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+# shellcheck disable=SC1091
+
+################################################################################
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+####################################################################################
+# This file is gathering all download boot jdk functions, so they can b reused later
+# On long run, the methods - due theirs simialrity - should converge to one
+####################################################################################
+
+# this one should be swithing per (supported) oses in future
+function downloadBootJDK() {
```

Comment:
Why are we not using this function now?

Reply:
I see it used  on https://github.com/adoptium/temurin-build/pull/3725/files#diff-1968af21c421fa9d7903dcd8950e640bc8ae257b7d7c5768c0fdd64d95fe3d7aR589

I had extracted two `downloadBootJDK` from `linux.sh` and `windows.sh` as `downloadLinuxBootJDK` and downloadWindowsBootJDK`. 
The original `linux.sh` is now calling `downloadLinuxBootJDK` and original `windows.sh` is calling `downloadWindowsBootJDK`

The "new" `downloadBootJDK` (maybe I shoud have change the name) is now called in `prepareWorkspace.sh` in `downloadBootJdkIfNeeded` 

`downloadBootJdkIfNeeded ` is decideing whether download should happenIf so, it is callin `downloader.sh`s downloadBootJDK` and that is decideding what platfrom to use.